setting.go

```diff
@@ -113,7 +123,9 @@ func EnableCookie(enable bool) {
 
 // SetTimeout sets the timeout for every request
 func (r *Req) SetTimeout(d time.Duration) {
-	r.Client().Timeout = d
+	client := newClient()
+	client.Timeout = d
+	r.SetClient(client)
```

Comment:
We should not create new client to override the old client, just modify the existed client. The Client() function ensures the client exists, and no race condition. So just using r.Client().Timeout = d is fine.

Reply:
You're right will change it.